tests/console/add_phub_extension.pm

```diff
@@ -11,18 +11,29 @@ use strict;
 use warnings;
 use testapi;
 use serial_terminal 'select_serial_terminal';
-use registration qw(add_suseconnect_product get_addon_fullname);
+use registration qw(add_suseconnect_product get_addon_fullname is_phub_ready);
 
 sub run {
     select_serial_terminal;
 
-    add_suseconnect_product(get_addon_fullname('phub'));
+    return unless is_phub_ready();
+
+    eval { add_suseconnect_product(get_addon_fullname('phub')); };
+    if ($@) {
+        if (check_var('BETA', '1')) {
+            force_soft_failure('poo#120879, PackageHub installation might fail in early development');
+            set_var('PHUB_READY', 0);
+        }
+        else {
+            die "$@";
+        }
+    }
```

Comment:
I don't understand this.

Reply:
Hmm,  I will have a try to activated phub, if it fails with `BETA=1`, then force fail the test